AzureIotPnpDps.X/main.c

```diff
@@ -7,582 +7,1327 @@
   } while (0)
 
+static int send_reported_property(twin_properties_t* twin_properties);
+
 extern az_iot_pnp_client pnp_client;
 extern pf_MQTT_CLIENT pf_mqtt_iotprovisioning_client;
 extern pf_MQTT_CLIENT pf_mqtt_pnp_client;
 
-// * Plug and Play Values *
-// The model id is the JSON document (also called the Digital Twins Model Identifier or DTMI)
-// which defines the capability of your device. The functionality of the device should match what
-// is described in the corresponding DTMI. Should you choose to program your own PnP capable device,
-// the functionality would need to match the DTMI and you would need to update the below 'model_id'.
-// Please see the sample README for more information on this DTMI.
-const az_span device_model_id = AZ_SPAN_LITERAL_FROM_STR("dtmi:com:example:Thermostat;1");
+// used by led.c to communicate LED states changes
+extern led_status_t led_status;
+
+// used by pin_manager.c
+extern button_press_data_t button_press_data;
+
+extern uint32_t telemetry_interval_seconds;
+
+static bool is_get_received = false;
+static int reboot_delay_seconds = 0;
+static uint32_t reboot_task(void* payload);
+static timerStruct_t reboot_timer = { reboot_task };
+
+#define LED_NO_CHANGE (-1)
+//
+// * LEDs *
+//        | On                   | Off                      | Fast Blink (100ms)           | Slow Blink (400ms)
+// Blue	  | AP Connected         | AP Disconnected          | WiFi Connectin in progress   | 
+// Green  | Connected to IoT Hub | Not Connected to IoT Hub | DPS provisioning in progress | IoT Hub connection in progress
+// Yellow | User LED Control from Cloud (On, Off, Blink)
+// Red    | Error                | No error                 | Error with DPS connection    | Error with IoT Hub connection
+//
+
+// * IoT Plug and Play Values *
+// The model id is the unique identifier for device model (also called the Digital Twins Model Identifier or DTMI)
+// The device Model defines the capability of your device. The functionality of the device should match what
+// is described in the corresponding device model while following IoT Plug and Play convension. 
+// Should you choose to program your own IoT Plug and Play device model,
+// the functionality would need to match the device model and you would need to update the below 'device_model_id'.
+// Please see the sample README for more information on IoT Plug and Play.
+const az_span device_model_id = AZ_SPAN_LITERAL_FROM_STR("dtmi:com:Microchip:PIC_IoT_WM;1");
+
 // ISO8601 Time Format
-static const char iso_spec_time_format[] = "%Y-%m-%dT%H:%M:%S.000%zZ";
+static const char iso_spec_time_format[] = "%Y-%m-%dT%H:%M:%SZ";
 
-// Plug and Play Connection Values
+// IoT Plug and Play Connection Values
 static int32_t request_id_int;
 static char request_id_buf[8];
 
-// Plug and Play Telemetry Values
+// IoT Plug and Play temperature and light sensor telemetry
+// Frequency of telemetry is determined by telemetryInterval desired property (or writable property)
```

Comment:
nit:
The language [here](https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/dtdlv2.md#property) makes little mention of "desired" and instead is writable, read-write, read-only, etc.
```suggestion
// Frequency of telemetry is determined by the writable property telemetryInterval.
```